### DISCUSSION TITLE
Get Media api returning record not found, but with same media ID, I can post that media.

### DISCUSSION THREAD

**Discussion by sudhagar-mr-17051** [mastodon/mastodon#24473] [Q&A] [answered]
Hi ,

I have uploaded a media via API and using the mediaID that I have got through response, I've create a post and then when I tried to fetch the media details using GET media method, it is return Record not found,

but I have used the same Bearer token for both posting the status and getting the media
The media is there, since the post has been published with that mediaID.

Please let me know if you need any further details

Kindly help me in resolving this,
Thanks in advance

**Comment by andypiper** [comment]
Can you share an example set of calls that demonstrate what you're describing?

**Reply by sudhagar-mr-17051** [reply]
for the mediaid: 110535522010168632

https://mastodon.social/api/v1/media/110535522010168632

If I publish an media and make get request to that media with obtained mediaid, I am getting empty response or getting response with null in url part. Is there any specific period of time for media upload, so that after that time we should try fetching it.And kindly share the common practice on statuses with media while using mastodon API

**Reply by andypiper** [reply]
Could you share a code snippet with the sequence of calls you are making, rather than just a media id?

Something like:

- call `/api/v2/media` to upload media
- get 200 OK or 202 Accepted
- call `/api/v1/media/:id` 
- get 200 OK or 206 Partial

At the moment it is difficult to follow what you are describing without any code example or walkthough.

"getting response with null in url part" -> this is explained [here](https://docs.joinmastodon.org/methods/media/#v1). It sounds like you're seeing the media being processed in the background.

**Reply by sudhagar-mr-17051** [reply]
called api/v2/media -> got 202 response code with json response that contains "remote_url" field as null

followedy by 

/api/v1/media/:id -> 206 code with same json
after few seconds
/api/v1/media/:id -> 206 code with same json
after few seconds
/api/v1/media/:id -> 206 code with same json

How long does it take usually for processing?
So that we can change the waiting time

**Reply by andypiper** [reply, marked as answer]
"usually" will depend on the instance: available capacity, size of processing queue, etc; and also on the size of the media. I don't know whether there is a specific recommendation for waiting time, you could try a incremental back-off approach (wait 2 seconds, then wait 4 seconds, then wait 8 seconds, and so on, by the time of the third check you've waited 14 seconds in total). If the media continues to take a long time to process, it could be an issue with processing backlog on the instance.

**Reply by sudhagar-mr-17051** [reply]
Thanks for the suggestion!
